PMSCore/Utils/UserManager.cs

```diff
@@ -0,0 +1,138 @@
+            _logger.LogDebug("Setting up User Manager.");
+
+            // Attempt login; if it fails, register a new user.
+            if (!this.Login())
+            {
+                this.RegisterNewUser();
+            }
+
+        }
+
+        /// <summary>
+        /// Retrieves the permission level of the current user.
+        /// </summary>
+        /// <returns>A string representing the user's permissions.</returns>
+        public string GetPermission()
+        {
+            return _currentUser.ViewUserPermissions();
+        }
+
+        /// <summary>
+        /// Retrieves the user ID of the current user.
+        /// </summary>
+        /// <returns>A string representing the user ID.</returns>
+        public string GetUserID()
+        {
+            return _currentUser.GetUserID();
+        }
+
+
+        /// <summary>
+        /// Simulates the login process. If no user is currently logged in, it creates a test user.
+        /// </summary>
+        /// <returns>True if login succeeds, false otherwise.</returns>
+        public bool Login()
+        {
+            if(_currentUser == null)
+            {
+                //Menu.DisplayLoginMenu();
+                //userID = PMSUtils.Sanitize(Console.NextLine());
+                //userpassword = PMSUtils.Sanitize(Console.NextLine());
+
+                Console.WriteLine("Simulating Login Process..." +
+                    "\n----------------------------------------------------------------------" +
+                    "\n To Login please enter your username, followed by your password." +
+                    "\nUsername: user1" +
+                    "\nPassword: *****" +
+                    "\n----------------------------------------------------------------------" +
+                    "\nUser would enter username and password here to be sent to back end.");
+
+                //*
+                // response = StateManager.ValidateUser(username, userpassword);
+
+                // if(response[0] == true && response[1] == true){
+                //      this._user = StateManager.LoadUserProfile(username);
+                //  }
+                //  elif(response[1] == false)
+                //  {
+                //      _logger.LogWarning("Invalid Password for User Account.{username}", username);
+                //  }
+                //  else
+                //  {
+                //      this.Register();
+                //  }
+                //*
+
+                // Simulates user login by creating a test user and assigning it a new Permission_Object.
+                _currentUser = new User_Tester
+                    (
+                    new LoggerFactory()
+                    .CreateLogger("User_Tester"),
+                    new Permission_Object(2),
+                    "J-Doe"
+                    );
+            }
+
+            return _currentUser != null; // Return true if user is successfully created.
+        }
```

Comment:
Login should receive a user object, if it exists in the list current_user becomes that user and returns a state else current_user remains what is was and an error status gets returned